src/components/TodoItem.tsx

```diff
@@ -0,0 +1,134 @@
+import classNames from 'classnames';
+import React, { useState, useEffect } from 'react';
+import { Todo } from '../types/Todo';
+import { client } from '../utils/fetchClient';
+
+export const TodoItem = React.memo(
+  ({
+    todo,
+    askTodos,
+    setErrorMessage,
+  }: {
+    todo: Todo,
+    askTodos: (url: string) => void
+    setErrorMessage: React.Dispatch<React.SetStateAction<string>>
+  }) => {
+    const { id, completed, title } = todo;
+    const [isComplited, setIsComplited] = useState(completed);
+    const [isEdit, setIsEdit] = useState(false);
+    const [temporaryText, setTemporaryText] = useState(title);
+    const [isLoading, setIsLoading] = useState(false);
+
+    useEffect(() => {
+      setIsComplited(completed);
+    }, [completed]);
+
+    const handleComplited = () => {
+      setIsLoading(prevState => !prevState);
+      setIsComplited(prevState => !prevState);
+
+      client.patch(`/todos/${id}`, { completed: !isComplited })
+        .finally(() => {
+          setIsLoading(prevState => !prevState);
+          askTodos('/todos?userId=6757');
+        })
```

Comment:
keep the then().catch().finally() structure

Reply:
https://github.com/mate-academy/react_todo-app-add-and-delete/pull/357#discussion_r1160509125
If I add useEffect in TodoInfo, my form not work correctly.